lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -204,7 +203,18 @@ def run!(verify_result: true, allow_protected_ns: false, prune: true)
     private
 
     def cluster_resource_discoverer
-      ResourceDiscovery.new(namespace: @namespace, context: @context, logger: @logger, namespace_tags: @namespace_tags)
+      @cluster_resource_discoverer ||= ResourceDiscovery.new(
+        namespace: @namespace,
+        context: @context,
+        logger: @logger,
+        namespace_tags: @namespace_tags
+      )
+    end
+
+    def add_dynamic_custom_resources_to_predeploy_sequence(resources)
+      crds = cluster_resource_discoverer.crds(@sync_mediator).map(&:kind)
+      discovered_crs = resources.select { |resource| crds.include? resource.type }
+      DeployTask.const_set("PREDEPLOY_SEQUENCE", (discovered_crs.map(&:type) + BASE_PREDEPLOY_SEQUENCE).uniq)
```

Comment:
Do we expect a case where the base crds would not be in the discovered set?

Reply:
On deploys that contain both the CRD and an instance of it as a resource. AFAICT we don't do that ourselves, but am unsure if the wider community that uses Kubernetes-deploy uses that pattern.

Reply:
we are only adding the CRs here not the CRD right?

Reply:
Updated to use the `prune_whitelist` approach`